l5kit/l5kit/dataset/ego.py

```diff
@@ -246,3 +253,242 @@ def _get_sample_function(self) -> Callable[..., dict]:
     def get_scene_dataset(self, scene_index: int) -> "EgoDatasetVectorized":
         dataset = self.dataset.get_scene_dataset(scene_index)
         return EgoDatasetVectorized(self.cfg, dataset, self.vectorizer, self.perturbation)
+
+
+class EgoAgentDatasetVectorized(EgoDatasetVectorized):
+    def __init__(
+        self,
+        cfg: dict,
+        zarr_dataset: ChunkedDataset,
+        vectorizer: Vectorizer,
+        perturbation: Optional[Perturbation] = None,
+        agents_mask: Optional[np.ndarray] = None,
+        eval_mode: bool = False
+    ):
+        """
+        Get a PyTorch dataset object that can be used to train DNNs with vectorized input.
+        Ego features are added to the agent features to treat ego as an agent.
+
+        Args:
+            cfg (dict): configuration file
+            zarr_dataset (ChunkedDataset): the raw zarr dataset
+            vectorizer (Vectorizer): a object that supports vectorization around an AV
+            perturbation (Optional[Perturbation]): an object that takes care of applying trajectory perturbations.
+                None if not desired
+            agents_mask (Optional[np.ndarray]): custom boolean mask of the agent availability.
+            eval_mode (bool): enable eval mode (iterates over agent, similarly to AgentDataset).
+        """
```

Comment:
Move this to the class docstring above (the ctor method is often not shown in the API docs, so this doc might be hidden).